src/main/java/de/tum/in/www1/artemis/service/ModelingSubmissionService.java

```diff
@@ -39,85 +40,148 @@ public ModelingSubmissionService(ModelingSubmissionRepository modelingSubmission
     }
 
     /**
-     * Saves the given submission and the corresponding model
+     * Saves the given submission and the corresponding model and creates the result if necessary.
+     * Furthermore, the submission is added to the AutomaticSubmissionService if not submitted yet.
+     * Is used for creating and updating modeling submissions.
+     * Rolls back if inserting fails - occurs for concurrent createModelingSubmission() calls.
+     *
      * @param modelingSubmission the submission to submit
      * @param modelingExercise the exercise to submit in
      * @param participation the participation where the result should be saved
      * @return the result entity
      */
-    @Transactional
+    @Transactional(rollbackFor = Exception.class)
     public Result save(ModelingSubmission modelingSubmission, ModelingExercise modelingExercise, Participation participation) {
-        // update submission properties
-        modelingSubmission.setType(SubmissionType.MANUAL);
+        User user = participation.getStudent();
+        if (modelingSubmission.getId() != null) {
+            // check if the the submission was already submitted due to auto submit
+            // and return the corresponding result if that's true
+            ModelingSubmission dbModelingSubmission = modelingSubmissionRepository.findOne(modelingSubmission.getId());
```

Comment:
Can we use web sockets instead to notify the client and not having such exception handling code